drivers/tc_client_data_report/app/models/tc_client_data_report/report.rb

```diff
@@ -0,0 +1,78 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module TcClientDataReport
+  class Report
+    include ::Filter::FilterScopes
+    include ArelHelper
+
+    def initialize(filter)
+      @filter = filter
+    end
+
+    def rows
+      enrollments = enrollment_scope.
+        preload(
+          :service_history_services,
+          service_history_enrollment_for_head_of_household: { enrollment: :income_benefits },
+          project: :project_cocs,
+        ).
+        where(client_id: client_scope.select(:id)).
+        order(first_date_in_program: :asc).
+        group_by(&:client_id)
+
+      client_scope.
+        map do |client|
+          enrollment = enrollments[client.id].first
+          program_name = enrollment.project.ProjectName
+          bed_night = enrollment.service_history_services.order(date: :asc).bed_night.first
+          hoh_income = enrollment.
+            service_history_enrollment_for_head_of_household.
+            enrollment.
+            income_benefits_at_entry.
+            TotalMonthlyIncome
+
+          {
+            program_name: program_name,
+            service_date: bed_night&.date || enrollment.first_date_in_program, # first bed night, or entry date
```

Comment:
These _should_ always be the same.  Not sure we need to go through the hoops of checking service history services.  If you aren't served on your first night, the enrollment shouldn't have been opened.

Instead, I think I'd limit the enrollment scope to those with service in the date range.